shell/common/common_TPC-H.sh

```diff
@@ -1,40 +1,73 @@
 # Common functions for different TPC-H implementations
 # Based on Hadoop and Hive
-
-source_file "$ALOJA_REPO_PATH/shell/common/common_hive.sh"
-set_hive_requires
-
-[ ! "$TPCH_SCALE_FACTOR" ] &&  TPCH_SCALE_FACTOR=1 #1 GB min size
-[ ! "$TPCH_USE_LOCAL_FACTOR" ] && TPCH_USE_LOCAL_FACTOR="" #set to a scale factor to use the local DBGEN instead of the M/R version
-
-BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes
-
+#
+if [[ "$NATIVE_FORMAT" == "text" && "$BENCH_SUITE" == *"native-spark"* ]]; then
+  source_file "$ALOJA_REPO_PATH/shell/common/common_hadoop.sh"
+  set_hadoop_requires
+else
+  source_file "$ALOJA_REPO_PATH/shell/common/common_hive.sh"
+  set_hive_requires
+fi
+
+if [ "$BB_SERVER_DERBY" == "true" ]; then
+  source_file "$ALOJA_REPO_PATH/shell/common/common_derby.sh"
+  set_derby_requires
+fi
+
+
+if [[ "$TPCH_USE_LOCAL_FACTOR" > 0 ]]; then
+  TPCH_SCALE_FACTOR=$TPCH_USE_LOCAL_FACTOR
+  BENCH_DATA_SIZE="$((TPCH_USE_LOCAL_FACTOR * 1000000000 ))" #in bytes
+else
+  [[ ! "$TPCH_SCALE_FACTOR" ]] && TPCH_SCALE_FACTOR=1 #1 GB min size
+  BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes
+fi
```

Comment:
Can we move `BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes` outside the conditional block, rather than repeating it?

Reply:
We are proposing a new solution for BENCH_DATA_SIZE to unify the way we enter bench sizes, I think it would be appropriate to leave it as it is for now as this very same lines will change eventually.